tests/test_docs_extraction.py

```diff
@@ -0,0 +1,335 @@
+    # inspect will fallback to the last found class in the source file.
+    # The following is to ensure using frames will still get the correct one
+    if True:
+
+        class MyModel(BaseModel):
+            a: int
+            """A docs"""
+
+            model_config = ConfigDict(
+                use_attribute_docstrings=True,
+            )
+
+    else:
+
+        class MyModel(BaseModel):
+            b: int
+            """B docs"""
+
+            model_config = ConfigDict(
+                use_attribute_docstrings=True,
+            )
+
+    assert MyModel.model_fields['a'].description == 'A docs'
+
+
+def test_model_docs_dedented_string():
+    # fmt: off
+    class MyModel(BaseModel):
+        def bar(self):
+            """
+An inconveniently dedented string
+            """
+
+        a: int
+        """A docs"""
+
+        model_config = ConfigDict(
+            use_attribute_docstrings=True,
+        )
+    # fmt: on
+    assert MyModel.model_fields['a'].description == 'A docs'
+
+
+def test_model_docs_inheritance():
+    class MyModel(BaseModel):
+        a: int
+        """A docs"""
+
+        b: int
+        """B docs"""
+
+        model_config = ConfigDict(
+            use_attribute_docstrings=True,
+        )
+
+    FirstModel = MyModel
+
+    class MyModel(FirstModel):
+        a: int
+        """A overridden docs"""
+
+    assert FirstModel.model_fields['a'].description == 'A docs'
+    assert MyModel.model_fields['a'].description == 'A overridden docs'
+    assert MyModel.model_fields['b'].description == 'B docs'
+
+
+def test_model_different_name():
+    # As we extract docstrings from cls in `ModelMetaclass.__new__`,
+    # we are not affected by `__name__` being altered in any way.
+
+    class MyModel(BaseModel):
+        a: int
+        """A docs"""
+
+        model_config = ConfigDict(
+            use_attribute_docstrings=True,
+        )
+
+    MyModel.__name__ = 'OtherModel'
+    print(MyModel.__name__)
+
+    assert MyModel.model_fields['a'].description == 'A docs'
+
+
+def test_model_generic():
+    class MyModel(BaseModel, Generic[T]):
+        a: T
+        """A docs"""
+
+        model_config = ConfigDict(
+            use_attribute_docstrings=True,
+        )
+
+    assert MyModel.model_fields['a'].description == 'A docs'
+
+    class MyParameterizedModel(MyModel[int]):
+        a: int
+        """A parameterized docs"""
+
+    assert MyParameterizedModel.model_fields['a'].description == 'A parameterized docs'
+    assert MyModel[int].model_fields['a'].description == 'A docs'
+
+
+def test_dataclass_no_docs_extraction():
+    @pydantic_dataclass
+    class MyModel:
+        a: int = 1
+        """A docs"""
+
+        b: str = '1'
+
+        """B docs"""
+
+    assert MyModel.__pydantic_fields__['a'].description is None
+    assert MyModel.__pydantic_fields__['b'].description is None
+
+
+def test_dataclass_docs_extraction():
+    @pydantic_dataclass(config=ConfigDict(use_attribute_docstrings=True))
```

Comment:
```suggestion
    @pydantic_dataclass(
        config=ConfigDict(use_attribute_docstrings=True),
    )
```

so that we can feel justified about the more complicated implementation :laughing: